tensorflow/core/kernels/dynamic_partition_op_gpu.cu.cc

```diff
@@ -337,13 +337,11 @@ class DynamicPartitionOpGPU : public AsyncOpKernel {
     // Determine temporary device storage requirements.
     Tensor cub_temp_storage;
     size_t temp_storage_bytes = 0;
-    auto status = gpuprim::DeviceRadixSort::SortPairs(
+    auto gpuResult = gpuprim::DeviceRadixSort::SortPairs(
                 NULL, temp_storage_bytes, partitions_ptr, partitions_out_ptr,
                 indices_in_ptr, indices_out_ptr, N, 0, sizeof(int32) * 8, cu_stream);
     
-    OP_REQUIRES_ASYNC(c, status,
-                    errors::Internal("Failed to launch copy from device to host."), 
-                    done);
+    if (gpuResult != gpuSuccess) return;    
```

Comment:
You are right, OP_REQUIRES_ASYNC does not work in this case. But it seems OP_REQUIRES would work:

OP_REQUIRES(c, gpuResult == gpuSuccess, errors::Internal("Failed to launch copy from device to host.");

This is better than just silently returning. Same in the other cases.

Reply:
Super! Yes indeed! Thanks for this suggestion. I have made the change please take another look. Thanks!